network/server.go

```diff
@@ -338,7 +339,11 @@ func (s *Server) checkPeerConnections() {
 
 		if s.numPeers() < MinimumPeerConnections {
 			if s.config.NoDiscover || !s.bootnodes.hasBootnodes() {
-				//TODO: dial peers from the peerstore
+				// dial unconnected peer
+				randPeer := s.GetRandomPeer()
+				if randPeer != nil && !s.IsConnected(*randPeer) {
+					s.addToDialQueue(s.GetPeerInfo(*randPeer), common.PriorityRandomDial)
+				}
 			} else {
 				randomNode := s.GetRandomBootnode()
```

Comment:
Should we check randomNode `nil`?

Reply:
@DarianShawn  It can return `nil` [network/server_discovery.go#L153](https://github.com/dogechain-lab/dogechain/blob/b6c3170392a2a5c45f636bd89a60d3826383c95f/network/server_discovery.go#L153)

Reply:
@DarianShawn fix miss code [fd1f824](https://github.com/dogechain-lab/dogechain/pull/214/commits/fd1f82415b90139e68e2712a7b015e41911e206a)